### PR TITLE
refactor docs with named data

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Stable Names for datasets](https://trello.com/c/zJVzRWzc/852-stable-names-for-datasets)

File: src/geoengine/datasets.md [hunk 1]
```diff
@@ -8,12 +8,19 @@ Geo Engine supports different types of data, reflected by a `DataId`, which refe
 
 An internal dataset is a dataset that is stored in the Geo Engine.
 Thus, it is efficiently accessible and can be used in workflows.
-The dataset is identified by a `DatasetId` and contains a `DatasetDefinition` that describes the data.
-The `DatasetId` is always a UUID.
+The dataset is identified by a `DatasetName` and contains a `DatasetDefinition` that describes the data.
+
+The `DatasetName` is a string that consists of a *namespace* (optional) and a *name*, separated by a colon.
```
Comment (michaelmattig) [inline src/geoengine/datasets.md:13]: maybe specify allowed characters in the name

File: src/geoengine/datasets.md [hunk 2]
```diff
@@ -8,12 +8,19 @@ Geo Engine supports different types of data, reflected by a `DataId`, which refe
...
 
 An external dataset is a dataset that is not stored in the Geo Engine.
 Geo Engine accesses it from a foreign location.
 The dataset is identified by an `ExternalDataId` that consists of a `DataProviderId` and a `LayerId`.
 While the `DatasetProviderId` is usually a UUID that identifies the data provider for Geo Engine itself, the `LayerId` is a string that identifies the layer in the data provider.
+
+The `ExternalDataId` is a string that consists of a *namespace*, the `DataProviderId` and a *name*, separated by a colon.
+The namespace cannot be omitted and is `_` for the global namespace.
+For instance, `_:{uuid}:name` or `namespace:{uuid}:name` refer to datasets.
+If the *name* is a complex string, it can be enclosed by backticks, e.g., ``namespace:{uuid}:`name with spaces` ``.
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply src/geoengine/datasets.md:26]: I can't get rid of it, unfortunately.

File: src/operators/gdalsource.md [hunk 3]
```diff
@@ -23,10 +23,7 @@ If the given dataset does not exist or is not readable, an error is thrown.
 {
   "type": "GdalSource",
   "params": {
-    "data": {
-      "type": "internal",
-      "datasetId": "a626c880-1c41-489b-9e19-9596d129859c"
-    }
+    "data": "ndvi"
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply src/operators/gdalsource.md:26]: [d954cf5](https://github.com/geo-engine/docs/pull/49/commits/d954cf5270f497b76e89ebfd0b45b59dfad5a022)

File: src/operators/ogrsource.md [hunk 4]
```diff
@@ -39,10 +39,7 @@ If the given dataset does not exist or is not readable, an error is thrown.
 {
   "type": "OgrSource",
   "params": {
-    "data": {
-      "type": "internal",
-      "datasetId": "e977b123-ca47-4c5b-aace-481119826aaf"
-    },
+    "data": "places",
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply src/operators/ogrsource.md:42]: [d954cf5](https://github.com/geo-engine/docs/pull/49/commits/d954cf5270f497b76e89ebfd0b45b59dfad5a022)

File: src/geoengine/datasets.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply src/geoengine/datasets.md]: [47e6597](https://github.com/geo-engine/docs/pull/49/commits/47e659797ab1ca35c973da28b63ae08d9a201b05)

**Pull request merged**